src/test/scala/io/flow/apidoc-json-validation/FormDataSpec.scala

```diff
@@ -22,6 +22,43 @@ class FormDataSpec extends FunSpec with Matchers {
     )
   }
 
+  it("rewriteEncoded") {
+    FormData.rewriteEncoded("number[]=100379876543&number[]=WT65xSPLX-SPT-5") should be(
+      "number=100379876543&number=WT65xSPLX-SPT-5"
+    )
+
+    FormData.rewriteEncoded("user[name][first]=mike&user[name][last]=bryzek") should be(
+      "user[name][first]=mike&user[name][last]=bryzek"
+    )
+
+    FormData.rewriteEncoded("user[name][first][]=mike&user[name][first][]=maciej&user[name][last]=bryzek") should be(
+      "user[name][first]=mike&user[name][first]=maciej&user[name][last]=bryzek"
+    )
+  }
+
+  it("arrays with []") {
+    FormData.parseEncoded("number[]=1&number[]=2") should be(
+      Map("number[]" -> Seq("1", "2"))
+    )
+
+    FormData.toJson(FormData.parseEncoded("number[]=1&number[]=2")) should be(
+      Json.obj("number" -> Seq("1", "2"))
+    )
+  }
+
+  it("arrays") {
+    FormData.parseEncoded("key=val1&key=val2") should be(
+      Map("key" -> Seq("val1", "val2"))
+    )
+    
+    FormData.parseEncoded("key=val1&key=val2&foo=bar") should be(
```

Comment:
Does this still work if the params are out of order, e.g. `key=val1&foo=bar&key=val2`?